numba/dictobject.py

```diff
@@ -166,6 +225,9 @@ def _sentry_safe_cast(fromty, toty):
         if isinstance(fromty, types.Integer) and isinstance(toty, types.Float):
             # Accept if ints to floats
             return
+        if isinstance(fromty, types.Float) and isinstance(toty, types.Float):
+            # Accept if floats to floats
+            return
```

Comment:
I think we need to revisit the typecast implementation to avoid these specialization.  We may need more `Conversion` enum values to fill the need for maybe-safe-interdomain-cast with runtime checking.  But, I will double-check the

Reply:
adde more tests for 
>  `int/float->complex` and `bool->anything` permitted?